core/src/main/scala/kafka/log/LogSegment.scala

```diff
@@ -346,14 +346,16 @@ class LogSegment private[log] (val log: FileRecords,
   @nonthreadsafe
   def truncateTo(offset: Long): Int = {
     val mapping = translateOffset(offset)
-    if (mapping == null)
-      return 0
     offsetIndex.truncateTo(offset)
     timeIndex.truncateTo(offset)
     txnIndex.truncateTo(offset)
     // after truncation, reset and allocate more space for the (new currently  active) index
     offsetIndex.resize(offsetIndex.maxIndexSize)
     timeIndex.resize(timeIndex.maxIndexSize)
+
+    if (mapping == null)
```

Comment:
There are three cases here:
1. The segment is not empty, but the target offset is larger than the last message in the segment.
2. The segment is not empty, but becomes empty after the truncation.
3. The segment is empty before the truncation.

It seems the original intention was to reset the create time if the segment becomes empty after truncation (case 2). The current patch does not do that for case 3. Do we want to maintain that behavior?

Reply:
Seems reasonable to reset the create time for case 3 as well.